shared/modules/selectors/networks.ts

```diff
@@ -43,6 +44,60 @@ export function getSelectedNetworkClientId(
 ) {
   return state.metamask.selectedNetworkClientId;
 }
+/**
+ * Type to extend InternalNetworkState with multichain configurations.
+ */
+type ExtendedInternalNetworkState = InternalNetworkState & {
+  multichainNetworkConfigurationsByChainId: Record<
+    string,
+    NetworkConfiguration
+  >;
+};
+
+/**
+ * State type that includes both standard and multichain network configurations.
+ */
+export type ConsolidatedNetworkConfigurationsState = {
+  metamask: Pick<
+    ExtendedInternalNetworkState,
+    | 'networkConfigurationsByChainId'
+    | 'multichainNetworkConfigurationsByChainId'
+  >;
+};
+
+/**
+ * Combines and returns network configurations for all chains (EVM and not).
+ *
+ * @param state - Redux state.
+ * @returns A consolidated object containing all available network configurations.
+ */
+export const getAllNetworkConfigurationsByCaipChainId = createDeepEqualSelector(
```

Comment:
this begs the question, should the NetworkController also keep evm networkconfigurations in caip2 format in the `multichainNetworkConfigurationsByChainId` in addition to the usual `networkConfigurationsByChainId` so that we can start migrating to this new property in state